cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceTaskContext.java

```diff
@@ -209,21 +209,32 @@ private void initializeTransactionAwares() {
     }
   }
 
-  /**
-   * This delegates the instantiation of the dataset to the super class, but in addition, if
```

Comment:
The piece of code starting the transaction is duplicate. So I extract them as a new function and move the comments there: https://github.com/caskdata/cdap/pull/6239/files#diff-aad06fed605dfd07f3806e9d07a5d3c2R228